pkg/kernel/Dockerfile

```diff
@@ -64,27 +65,11 @@ RUN case $(uname -m) in \
         fi && \
         cat /kernel_config${EXTRA} >> ${KERNEL_DEF_CONF}; \
     fi && \
-    make defconfig && \
-    make oldconfig && \
-    if [ -z "${EXTRA}" ]; then diff -cw .config ${KERNEL_DEF_CONF}; fi
+    cp ${KERNEL_DEF_CONF} .config && \
+    cp ${KERNEL_DEF_CONF} .config.new && \
+    make prepare
```

Comment:
I don't understand the implications of this change.

Today if I edit in a new option in the kernel config, then the diff will fail of some additional items are now available to be set. This forces us to maintain config files which list all of optional items, which makes it slightly easier to notice when something might be missing.

How will that editing in of a new option work with this new approach?

Reply:
There is no new approach, I just tried to avoid creepy if/else branch:
```
    x86_64) \
        KERNEL_DEF_CONF=/linux/arch/x86/configs/x86_64_defconfig; \
        ;; \
    aarch64) \
        KERNEL_DEF_CONF=/linux/arch/arm64/configs/defconfig; \
        ;; \
        ....
        cp ${KERNEL_DEF_CONF} .config.new && \
```

this KERNEL_DEF_CONF is defined in one RUN instance, another RUN instance does not see the variable defined in another RUN instance, so I just copy it to .config.new in order to access it between RUN instances. Simple and does not require variables, if branches, etc

Reply:
And forgot to mention: we need to prepare a .config in the kernel tree in order to prepare zfs built-in. So ok, I admit procedure is a bit different:
1. put .config into the kernel tree, run make prepare in order to prepare zfs built-in.
2. run zfs ./configure, prepare built-in, copy-builtin
3. verify configs 
4. make kernel (with zfs, etc)

Reply:
ok